components/AudioPlayer.tsx

```diff
@@ -0,0 +1,388 @@
+import { Box, IconButton } from "@mui/material";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const AudioPlayer = () => {
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [currentTime, setCurrentTime] = useState(0);
+  const [activeVolume, setActiveVolume] = useState(false);
+  const [isFirstRender, setIsFirstRender] = useState(true);
+  const [volume, setVolume] = useState(100);
+
+  const audioPlayer = useRef<HTMLAudioElement>(null);
+  const progressBar = useRef<HTMLInputElement>(null);
+  const volumeBar = useRef<HTMLInputElement>(null);
+
+  const pageInitialized = useRef(false);
+
+  const animationRef = useRef(0);
```

Comment:
What this zero value represents?

Reply:
This ref get the number type and the default number equals a zero

Reply:
@fordelord then change the variable name, so it is more obvious what kind of value it has

the word `animate` does not hint into what kind of variable it is